climpred_skeleton/comparison.py

```diff
@@ -0,0 +1,95 @@
+from typing import Union
+
+import pkg_resources
+import xarray as xr
+import yaml
+
+from .core import Verification
+
+
+class Comparison(Verification):
+    """First step in pipeline. Broadcast your prediction and observations
+    in a certain manner.
+
+    Base Comparison class hints at which attributes are expected for each
+    sub comparison class. It also stores the factory for internal use, to
+    return the appropriate subfunction.
+    """
+
+    _comparison_type: str
+    _hindcast_comparison: bool
+    _probabilistic_comparison: bool
+
+    def __init__(self, prediction: xr.Dataset, observation: xr.Dataset) -> None:
+        super().__init__(prediction, observation)
+
+    def factory(self, type) -> 'Comparison':
+        try:
+            return comparison_dict[type](self._prediction, self._observation)
+        except KeyError:
+            raise ValueError(
+                f'{type} not valid keyword from {list(comparison_dict.keys())}'
+            )
+
+
+class e2o(Comparison):
```

Comment:
Since this is a class, not sure if it should follow CamelCase e.g. `E2O`

Reply:
I'm not sure camel case is even supposed to have numbers. Maybe it should be more explicit, like `EnsembleToObservations`?